ground/src/main/java/com/google/android/ground/model/locationofinterest/LocationOfInterest.kt

```diff
@@ -70,50 +72,22 @@ sealed class LocationOfInterest {
             .build()
     }
 
-    // TODO: Remove once all callers are converted to Kotlin. We only retain this for Java interop.
-    open class Builder {
-        var id: String = ""
-            @JvmSynthetic set
-        var survey: Survey? = null
-            @JvmSynthetic set
-        var job: Job? = null
-            @JvmSynthetic set
-        var customId: String? = null
-            @JvmSynthetic set
-        var caption: String? = null
-            @JvmSynthetic set
-        var created: AuditInfo? = null
-            @JvmSynthetic set
-        var lastModified: AuditInfo? = null
-            @JvmSynthetic set
+    // TODO: These values exist for compatibility and can be removed once we eliminate the Point class.
+    val coordinatesAsPoints: ImmutableList<Point> =
+        geometry.coordinates.map(Point::fromCoordinate).toImmutableList()
 
-        open fun setId(value: String): Builder = apply { this.id = value }
-        open fun setSurvey(value: Survey): Builder = apply { this.survey = value }
-        open fun setJob(value: Job): Builder = apply { this.job = value }
-        open fun setCustomId(value: String?): Builder = apply { this.customId = value }
-        open fun setCaption(value: String?): Builder = apply { this.caption = value }
-        open fun setCreated(value: AuditInfo): Builder = apply { this.created = value }
-        open fun setLastModified(value: AuditInfo): Builder = apply { this.lastModified = value }
-    }
-
-    open fun builder(init: Builder.() -> Unit): Builder {
-        val builder = Builder()
-        builder.init()
-        return builder
-    }
+    val coordinatesAsPoint: Point =
+        Point.fromCoordinate(geometry.coordinate)
 
-    open fun toBuilder() = builder {
-        id = this@LocationOfInterest.id
-        survey = this@LocationOfInterest.survey
-        job = this@LocationOfInterest.job
-        customId = this@LocationOfInterest.customId
-        caption = this@LocationOfInterest.caption
-        created = this@LocationOfInterest.created
-        lastModified = this@LocationOfInterest.lastModified
-    }
-
-    companion object {
-        @JvmStatic
-        fun newBuilder(): Builder = Builder()
-    }
+    // "Lens" style setter for geometries.
+    fun <T : Geometry> updateGeometry(geometry: T): LocationOfInterest = LocationOfInterest(
```

Comment:
Would this create a new instance of `LocationOfInterest` every time the `geometry` is updated? Is it not possible to update the property from the caller place?

Reply:
Yes that's right, it always returns a new instance. This is generally how you update immutable structures in functional programming. The big difference here is that functional languages usually use sharing under the hood to mitigate cost (so even though it looks like an immutable copy, it's implemented as mutation under the hood).

I'm not sure whether or not Kotlin has similar optimizations, we'd have to look at the bytecode. Instead, I can change this to a `var` and make the setter private to give us immutability without the additional allocation costs.